### ISSUE TITLE
test_fencing fails in debug mode: Not enough data for satisfy transfer length header

### ISSUE THREAD

**Issue by avikivity** [scylladb/scylladb#18131]
https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7822/testReport/junit/(root)/non-boost%20tests/topology_experimental_raft_test_fencing_debug_1/

```

=================================== FAILURES ===================================
_______________________________ test_fence_hints _______________________________

self = <aiohttp.client_proto.ResponseHandler object at 0x7f9eb441c3d0>
exc = ClientPayloadError('Response payload is not completed')

    def connection_lost(self, exc: Optional[BaseException]) -> None:
        self._drop_timeout()
    
        if self._payload_parser is not None:
            with suppress(Exception):
                self._payload_parser.feed_eof()
    
        uncompleted = None
        if self._parser is not None:
            try:
>               uncompleted = self._parser.feed_eof()

/usr/lib64/python3.11/site-packages/aiohttp/client_proto.py:83: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   aiohttp.http_exceptions.TransferEncodingError: 400, message:
E     Not enough data for satisfy transfer length header.

aiohttp/_http_parser.pyx:507: TransferEncodingError

The above exception was the direct cause of the following exception:

request = <FixtureRequest for <Function test_fence_hints>>
manager = <test.pylib.manager_client.ManagerClient object at 0x7f9eb663d750>

    @pytest.mark.asyncio
    @skip_mode('release', 'error injections are not supported in release mode')
    async def test_fence_hints(request, manager: ManagerClient):
        logger.info("Bootstrapping cluster with three nodes")
        s0 = await manager.server_add(config={
            'error_injections_at_startup': ['decrease_hints_flush_period']
        }, cmdline=['--logger-log-level', 'hints_manager=trace'])
    
        # Disable load balancer as it might bump topology version, potentially creating a race condition
        # with read modify write below.
        # This should be done before adding the last two servers,
        # otherwise it can break the version == fence_version condition
        # which the test relies on.
        await manager.api.disable_tablet_balancing(s0.ip_addr)
    
        [s1, s2] = await manager.servers_add(2)
    
        logger.info(f'Creating test table')
        random_tables = RandomTables(request.node.name, manager, unique_name(), 3)
        table1 = await random_tables.add_table(name='t1', pks=1, columns=[
            Column("pk", IntType),
            Column('int_c', IntType)
        ])
        cql = manager.get_cql()
        await cql.run_async(f"USE {random_tables.keyspace}")
    
        logger.info(f'Waiting for cql and hosts')
        hosts = await wait_for_cql_and_get_hosts(cql, [s0, s2], time.time() + 60)
    
        host2 = host_by_server(hosts, s2)
        new_version = (await get_version(manager, host2)) + 1
        logger.info(f"Set version and fence_version to {new_version} on node {host2}")
        await set_version(manager, host2, new_version)
        await set_fence_version(manager, host2, new_version)
    
        select_all_stmt = SimpleStatement("select * from t1", consistency_level=ConsistencyLevel.ONE)
        rows = await cql.run_async(select_all_stmt, host=host2)
        assert len(list(rows)) == 0
    
        logger.info(f"Stopping node {host2}")
        await manager.server_stop_gracefully(s2.server_id)
    
        host0 = host_by_server(hosts, s0)
        logger.info(f"Writing through {host0} to regular column")
        await cql.run_async("insert into t1(pk, int_c) values (1, 1)", host=host0)
    
        logger.info(f"Starting last node {host2}")
        await manager.server_start(s2.server_id)
    
        logger.info(f"Waiting for failed hints on {host0}")
        async def at_least_one_hint_failed():
            metrics_data = await manager.metrics.query(s0.ip_addr)
            if sent_metric(metrics_data) > 0:
                pytest.fail(f"Unexpected successful hints; metrics on {s0}: {all_hints_metrics(metrics_data)}")
            if send_errors_metric(metrics_data) >= 1:
                return True
            logger.info(f"Metrics on {s0}: {all_hints_metrics(metrics_data)}")
>       await wait_for(at_least_one_hint_failed, time.time() + 60)

test/topology_experimental_raft/test_fencing.py:174: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test/pylib/util.py:49: in wait_for
    res = await pred()
test/topology_experimental_raft/test_fencing.py:168: in at_least_one_hint_failed
    metrics_data = await manager.metrics.query(s0.ip_addr)
test/pylib/rest_client.py:412: in query
    data = await self.client.get_text('/metrics', host=server_ip)
test/pylib/rest_client.py:82: in get_text
    ret = await self._fetch("GET", resource_uri, response_type = "text", host = host,
test/pylib/rest_client.py:72: in _fetch
    return await getattr(resp, response_type)()
/usr/lib64/python3.11/site-packages/aiohttp/client_reqrep.py:1143: in text
    await self.read()
/usr/lib64/python3.11/site-packages/aiohttp/client_reqrep.py:1101: in read
    self._body = await self.content.read()
/usr/lib64/python3.11/site-packages/aiohttp/streams.py:373: in read
    block = await self.readany()
/usr/lib64/python3.11/site-packages/aiohttp/streams.py:395: in readany
    await self._wait("readany")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <StreamReader e=ClientPayloadError('Response payload is not completed')>
func_name = 'readany'

    async def _wait(self, func_name: str) -> None:
        # StreamReader uses a future to link the protocol feed_data() method
        # to a read coroutine. Running two read coroutines at the same time
        # would have an unexpected behaviour. It would not possible to know
        # which coroutine would get the next data.
        if self._waiter is not None:
            raise RuntimeError(
                "%s() called while another coroutine is "
                "already waiting for incoming data" % func_name
            )
    
        waiter = self._waiter = self._loop.create_future()
        try:
            with self._timer:
>               await waiter
E               aiohttp.client_exceptions.ClientPayloadError: Response payload is not completed

/usr/lib64/python3.11/site-packages/aiohttp/streams.py:302: ClientPayloadError
```

**Comment by kbr-scylla** [comment]
Duplicate of https://github.com/scylladb/scylladb/issues/18083
